lib/ansible/modules/cloud/amazon/ec2_eni.py

```diff
@@ -501,7 +509,7 @@ def uniquely_find_eni(connection, module):
             filters['private-ip-address'] = private_ip_address
             filters['subnet-id'] = subnet_id
 
-        if instance_id and device_index:
+        if not attached and instance_id and device_index:
```

Comment:
If attached is specified the attachment status should be allowed to change as the docs state. Filtering by an instance_id and device_index that have yet to be attached is unexpected behavior.